examples/digital_fingerprinting/production/morpheus/dfp/modules/dfp_rolling_window.py

```diff
@@ -129,6 +130,10 @@ def build_window(message: MessageMeta, user_id: str) -> MessageMeta:
             return MessageMeta(cudf.from_pandas(train_df))
 
     def on_data(control_message: MessageControl):
+
+        if not control_message.has_task(task_type):
+            return None
```

Comment:
I don't think we want to be quite this strict. The rolling window stage could just receive empty CMs with a payload and a user_id, and add them to the cache. I think you're right in the sense that we need 'task_type', unless we say rolling window is only for training (which I think might be justified), so it can create CMs correctly.

I also want to be a bit careful with any path that discards CM without processing them; it should probably be an error to discard a CM with tasks on it, or various other criteria.